plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/launcher/DefaultedPipelineOptionsComponent.java

```diff
@@ -136,9 +139,21 @@ private void setWidgetsEnabled(boolean enabled) {
     defaultOptions.setEnabled(enabled);
   }
 
+  private String getCurrentEmail() {
+    IGoogleApiFactory apiFactory = new GoogleApiFactory();
+    if (apiFactory.hasCredentialsSet()) {
+      try {
+        return apiFactory.getAccount().getEmail();
+      } catch (IOException ex) {
+        // will return default
+      }
+    }
+    return "";
+  }
+  
   public Map<String, String> getValues() {
     Map<String, String> values = new HashMap<>();
-    values.put(DataflowPreferences.ACCOUNT_EMAIL_PROPERTY, defaultOptions.getAccountEmail());
+    values.put(DataflowPreferences.ACCOUNT_EMAIL_PROPERTY, getCurrentEmail());
```

Comment:
This is where it defaults to the ADC email regardless if using custom preferences or default ones